### ISSUE TITLE
Add `empty_enum_variants_with_brackets`

### ISSUE THREAD

**Issue by 0xLucqs** [software-mansion/cairo-lint#12]
# What it does
Finds enum variants without fields that are declared and used with empty brackets.

# Why restrict this?
Empty brackets after a enum variant declaration are redundant and can be omitted, and it may be desirable to do so consistently for style.

# Example

```rs
enum MyEnum {
   HasData: u8,
   HasNoData: ()       // redundant parentheses
}

let a = MyEnum::HasNoData(()); // redundant parentheses
```

Use instead:

```rs
enum MyEnum {
   HasData(u8),
   HasNoData,
}

let a = MyEnum::HashNoData;
```
https://rust-lang.github.io/rust-clippy/master/index.html#/empty_enum_variants_with_brackets

**Comment by addegbenga** [comment]
Can i work on this

**Comment by 0xLucqs** [comment]
yes and please open a PR ASAP so i can track where you are and can help you more easily

**Comment by guha-rahul** [comment]
Can i work on this ?

**Comment by raizo07** [comment]
@0xLucqs it's now you're supposed to assign me here. I'll be pushing the PR like I mentioned today

**Comment by MariangelaNM** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
Hi @0xLucqs can I work on this?

**Comment by adpthegreat** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
I am a rust developer of close to a year that is interested in open source and tackling problems   I recently made contributions to the technical content of solana-developers and I aim looking to make more meaningful contributions to other rust based projects in the web3 ecosystem 

### How I plan on tackling this issue
I would first read through the codebase and look at how other lints were implemented to have an idea of how I can model my solution , then create the file for the lint and write the appropriate logic that matches the lint rules

**Comment by SumitGoulikar** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/cairo-lint).

### My background and how it can be leveraged
I'm an undergraduate in Computer Science and am well experienced in coding across different programming languages. Along the way, I have enjoyed building efficient algorithms, practicing data structures, and tackling problems in clean, optimized code. So far, I've had fun using best practices to enhance performance and readability in projects.

I'm excited about this project because it will take my coding abilities, problem-solving skills, and passion for learning to the benefit of improving and maintaining code quality. With my strong foundation in computer science and eagerness to collaborate on impactful projects, you can expect to receive valuable contributions from me.
